### PR TITLE
prototype composition

### PR DESCRIPTION
**Pull request by PaulRitter**
requires https://github.com/space-wizards/space-station-14/pull/7403

still requires a check to look for circular composition

### CODE REVIEW

File: Robust.Server/Bql/BqlQuerySelector.Builtin.cs [hunk 1]
```diff
@@ -268,7 +268,7 @@ public override IEnumerable<EntityUid> DoSelection(IEnumerable<EntityUid> input,
                 if ((metaData.EntityPrototype?.ID == name) ^ isInverted)
                     return true;
 
-                return (metaData.EntityPrototype?.Parent == name) ^ isInverted; // Damn, can't actually do recursive check here.
+                return metaData.EntityPrototype?.Parents?.Any(x => x == name) == true ^ isInverted; // Damn, can't actually do recursive check here.
```
Comment (PaulRitter) [inline Robust.Server/Bql/BqlQuerySelector.Builtin.cs:271]: @moonheart08 unsure if this is correct

File: Robust.Shared/Localization/LocalizationManager.Entity.cs [hunk 2]
```diff
@@ -125,10 +126,10 @@ private EntityLocData CalcEntityLoc(string prototypeId)
                     }
                 }
 
-                if (prototype?.Parent == null)
+                if (prototype?.Parents == null)
                     break;
 
-                prototypeId = prototype.Parent;
+                prototypeId = prototype.Parents.First();
```
Comment (PaulRitter) [inline Robust.Shared/Localization/LocalizationManager.Entity.cs:129]: @RemieRichards unsure if this is correct or how to handle this

File: Robust.Server/Bql/BqlQuerySelector.Builtin.cs (hunk 1 above)
Comment (moonheart08) [reply Robust.Server/Bql/BqlQuerySelector.Builtin.cs:271]: This is correct.

**Comment by metalgearsloth** [comment]
Well that master merge went well

**Pull request closed without merging**